### PR TITLE
Add tests for long names

### LINKED ISSUE #107: Need tests around string limits
The CSI spec says that strings should not exceed 128 bytes and string maps should not exceed 4096 bytes. It's not clear where this enforcement should happen, but it would be nice to have tests that validate that plugin don't return strings that exceed these limits and negative tests that exceed these limits.

More importantly, we need tests that create volumes/snapshots using the maximum length names, to catch plugins which are using the name somewhere that can't handle 128 bytes.

### PR DESCRIPTION
**Pull request by bswartz**
Create 2 new tests, one for a volume with a maximum-length name,
and one for a snapshot with maximum-length name. These tests are
intended to catch bugs where plugin authors use the value of the
name field in a context where 128 characters exceeds a limit in
some other system.

Fixes #107

### CODE REVIEW

File: pkg/sanity/controller.go [hunk 1]
```diff
@@ -443,6 +445,56 @@ var _ = DescribeSanity("Controller Service", func(sc *SanityContext) {
 			Expect(err).NotTo(HaveOccurred())
 			cl.UnregisterVolume(name)
 		})
+
+		It("should not fail when creating volume with maximum-length name", func() {
```
[the remark this answers is not in the document]
Comment (pohly) [reply pkg/sanity/controller.go]: I think this is debatable. The spec makes a guarantee to the CSI driver that the name length won't exceed 128 bytes. It's then reasonable to assume that CSI drivers support names up to that size. Otherwise how can a CO know how long it can make the names without breaking some CSI driver which happens to support only shorter names? Therefore I think this test is okay.

[the remark this answers is not in the document]
Comment (lpabon) [reply pkg/sanity/controller.go]: In a driver implementation it is ok, but in csi-sanity we cannot assume any situation. This PR is making a clear decision that it wants the driver to support 128 chars. That is not in the spec.

[the remark this answers is not in the document]
Comment (lpabon) [reply pkg/sanity/controller.go]: @pohly Let me know that I was talking about the ID. My mistake. This is a good test.
